feat(slider) migrate Slider-component

albinscania opened the pull request:
**Describe pull-request**  
Migrated Slider component

**Solving issue**  
Fixes: #2416

**How to test**  
Check out the branch and click around.
Note that the eventlistener example will only work in canvas mode

mJarsater commented on tegel/src/components/slider/slider.stories.ts:
```diff
@@ -0,0 +1,265 @@
+      padding-right: 40px;
+    }
+    .demo-wrapper.sdds-on-white-bg {
+      background-color: #fff;
```
I think this can be removed, since we have the overall theme controls in place in another PR.

mJarsater commented on tegel/src/components/slider/slider.stories.ts:
```diff
@@ -0,0 +1,265 @@
+        type: 'boolean',
+      },
+    },
+    colorTheme: {
```
I couldn't see that this is part of the actual components so I think this can be removed, since we have the overall theme controls in place in another PR.

mJarsater commented on tegel/src/components/slider/slider.stories.ts:
```diff
@@ -0,0 +1,265 @@
+        type: 'boolean',
+      },
+    },
+    showTickNumbers: {
```
Maybe this one and showTicks should be dependant on each other and only show if the other one is true, not sure which should be the decisive one though.

albinscania commented:
I've added conditional controls, so it only shows the relevant controls. Also added the guards for observer/eventlisteners and a TODO remark for the reset() method (as in the 4.10 component PR)
